test/builtin/files_io/test_files.py

```diff
@@ -80,6 +80,262 @@ def test_close():
     ), f"temporary filename {temp_filename} should not appear"
 
 
+@pytest.mark.parametrize(
+    ("str_expr", "msgs", "str_expected", "fail_msg"),
+    [
+        ('Close["abc"]', ("abc is not open.",), "Close[abc]", None),
+        (
+            "exp = Sin[1]; FilePrint[exp]",
+            ("File specification Sin[1] is not a string of one or more characters.",),
+            "FilePrint[Sin[1]]",
+            None,
+        ),
+        (
+            'FilePrint["somenonexistentpath_h47sdmk^&h4"]',
+            ("Cannot open somenonexistentpath_h47sdmk^&h4.",),
+            "FilePrint[somenonexistentpath_h47sdmk^&h4]",
+            None,
+        ),
+        (
+            'FilePrint[""]',
+            ("File specification  is not a string of one or more characters.",),
+            "FilePrint[]",
+            None,
+        ),
+        (
+            'Get["SomeTypoPackage`"]',
+            ("Cannot open SomeTypoPackage`.",),
+            "$Failed",
+            None,
+        ),
+        ## Parser Tests
+        (
+            "Hold[<< ~/some_example/dir/] // FullForm",
+            None,
+            'Hold[Get["~/some_example/dir/"]]',
+            None,
+        ),
+        (
+            r"Hold[<<`/.\-_:$*~?] // FullForm",
+            None,
+            r'Hold[Get["`/.\\\\-_:$*~?"]]',
+            None,
+        ),
+        (
+            "OpenRead[]",
+            ("OpenRead called with 0 arguments; 1 argument is expected.",),
+            "OpenRead[]",
+            None,
+        ),
+        (
+            "OpenRead[y]",
+            ("File specification y is not a string of one or more characters.",),
+            "OpenRead[y]",
+            None,
+        ),
+        (
+            'OpenRead[""]',
+            ("File specification  is not a string of one or more characters.",),
+            "OpenRead[]",
+            None,
+        ),
+        (
+            'OpenRead["MathicsNonExampleFile"]',
+            ("Cannot open MathicsNonExampleFile.",),
+            "OpenRead[MathicsNonExampleFile]",
+            None,
+        ),
+        (
+            'fd=OpenRead["ExampleData/EinsteinSzilLetter.txt", BinaryFormat -> True, CharacterEncoding->"UTF8"]//Head',
```

Comment:
@mmatera The sequence of tests from here to line 172 should be put in a separate test. Subsequent tests depend on the actions of earlier tests.  If an early test fails, there is no cleanup. I am seeing for example "MathicsNonExampleFIle" appearing in the file system on failure and equally worse, in the mathics-core directory which should not be touched by testing. 

In that new separate test we should be doing cleanup when something goes wrong or a test fails. Context managers can help, or possibly the use of a temporary new temporary file or directory. 

For tests that are writing files, I do not think we are ensuring that we have access to a writable directory.